addons/l10n_latam_check/models/account_payment.py

```diff
@@ -337,3 +337,17 @@ def _create_paired_internal_transfer_payment(self):
                     default_l10n_latam_check_id=rec.l10n_latam_check_id,
                 ))._create_paired_internal_transfer_payment()
         super(AccountPayment, self - third_party_checks)._create_paired_internal_transfer_payment()
+
+    @api.constrains('l10n_latam_check_id')
+    def _check_l10n_latam_check_id(self):
+        if self.filtered(lambda x: x.payment_method_line_id.code == 'out_third_party_checks'):
+            payments = self.env['account.payment'].search([
```

Comment:
```suggestion
            payments = self.env['account.payment'].search_count([
```
search_count is less costly bc it doesn't require transmitting the records back to Python, and doesn't do an ordering (which search does by default)